Our collaborative network includes a number of groups responding to this RFA. This group of investigators came together around a set of responses to the RFA that were shared openly on GitHub as they were written. Selected interactions between groups include:

* _Augmented training_ is a technique used in deep learning applications to image data. During training, arbitrary patches are selected or arbitrary rotations are applied to training data to avoid learning irrelevant structure in data. We propose an innovative equivalent for single cell expression data by sampling reads. This requires fast expression quantification. We plan to work with Rob Patro, an expert in fast gene expression quantification, to implement data augmentation by very rapidly single cell observations.
* _Deep learning_ is expected to be relevant to other proposals. For example, Arjun Raj proposes to apply these methods for image segmentation analysis. We recently completed a large GitHub-authored collaborative review to identify deep learning strategies that address biological problems. We look forward to advising Arjun's group and others as they apply these methods.
* _Benchmark data_ will be something that we provide, and also we will benefit from the benchmark data of others. We contribute a large compendium of bulk gene expression data from multiple rheumatic diseases. This, combined with HCA single cell assays, will help us to evaluate the extent to which bulk public data measuring disease states can be explained by HCA-identified cell types. Our analysis of latent spaces also benefits from datasets that measure parallel perturbations across multiple cell types, as proposed by Arjun Raj. Such data from the Raj lab or other groups will assist us in evaluating our unsupervised deep learning methods in this context.
* _Model Interpretation_ is a key factor for any unsupervised approach. Elana Fertig and Loyal Goff's applications (shared via GitHub) describes methods for the interpretation of unsupervised models. We are excited to collaborate on methods and infrastructure for model interpretation. We have previously interacted via github on related work and added features based on between-group requests.
* _Interactive analyses_ can empower bench scientists to direct the analysis of single-cell genomics data. Lana Garmire's group has developed Granatum, an interactive graphic interface for single cell analysis and is responding to this analyses. We will provide source code for generating VAEs that can be incorporated into Granatum and Granatum-like systems.
* _Reproducible workflows_ will be needed to consistently apply and evaluate methods. We are happy to work with any participating groups, as well as CZI software developers, to provide and configure continuous integration servers for scientific workflows. For example, Rob Patro's group has previously extended our example configurations to provide a workflow for Salmon. We have not budgeted this in our application as a centrally provided service makes sense; however, we can modify our budget to cover this if requested.

If you are writing a proposal that you feel is relevant to ours, please file a pull request to add a mention or let me know in some way.
